tests/phpunit/tests/abilities-api/wpCoreAbilities.php

```diff
@@ -0,0 +1,135 @@
+<?php declare( strict_types=1 );
+
+/**
+ * Tests for the core abilities shipped with the Abilities API.
+ *
+ * @group abilities-api
+ */
+class Tests_Abilities_API_WpCoreAbilities extends WP_UnitTestCase {
+
+	/**
+	 * Tests that the `core/get-site-info` ability is registered with the expected schema.
+	 */
+	public function test_core_get_bloginfo_ability_is_registered(): void {
+		$ability = wp_get_ability( 'core/get-site-info' );
+
+		$this->assertInstanceOf( WP_Ability::class, $ability );
+		$this->assertTrue( $ability->get_meta_item( 'show_in_rest', false ) );
+
+		$input_schema  = $ability->get_input_schema();
+		$output_schema = $ability->get_output_schema();
+
+		$this->assertSame( 'object', $input_schema['type'] );
+		$this->assertArrayHasKey( 'default', $input_schema );
+		$this->assertSame( array(), $input_schema['default'] );
+
+		// Input schema should have optional fields array.
+		$this->assertArrayHasKey( 'fields', $input_schema['properties'] );
+		$this->assertSame( 'array', $input_schema['properties']['fields']['type'] );
+		$this->assertContains( 'name', $input_schema['properties']['fields']['items']['enum'] );
+
+		// Output schema should have all fields documented.
+		$this->assertArrayHasKey( 'name', $output_schema['properties'] );
+		$this->assertArrayHasKey( 'url', $output_schema['properties'] );
+		$this->assertArrayHasKey( 'version', $output_schema['properties'] );
+	}
+
+	/**
+	 * Tests executing the `core/get-site-info` ability returns all fields by default.
+	 */
+	public function test_core_get_bloginfo_executes(): void {
+		// Requires manage_options.
+		$admin_id = self::factory()->user->create( array( 'role' => 'administrator' ) );
+		wp_set_current_user( $admin_id );
+
+		$ability = wp_get_ability( 'core/get-site-info' );
+
+		// Test without fields parameter - should return all fields.
+		$result = $ability->execute();
+
+		$this->assertIsArray( $result );
+		$this->assertArrayHasKey( 'name', $result );
+		$this->assertArrayHasKey( 'description', $result );
+		$this->assertArrayHasKey( 'url', $result );
+		$this->assertArrayHasKey( 'version', $result );
+		$this->assertSame( get_bloginfo( 'name' ), $result['name'] );
+
+		// Test with fields parameter - should return only requested fields.
+		$result = $ability->execute(
+			array(
+				'fields' => array( 'name', 'url' ),
+			)
+		);
+
+		$this->assertIsArray( $result );
+		$this->assertArrayHasKey( 'name', $result );
+		$this->assertArrayHasKey( 'url', $result );
+		$this->assertArrayNotHasKey( 'description', $result );
```

Comment:
Nit: maybe checking the count of fields would be simpler instead of looking for all missing keys.